### ISSUE TITLE
Panic in go-memdb on arm64 darwin

### ISSUE THREAD

**Issue by tristanmorgan** [hashicorp/nomad#13804]
### Nomad version
`Nomad v1.3.2 (bf602974112964e9691729f3f0716ff2bcdb3b44)`

### Operating system and Environment details
macOS 12.4 
on M1 (arm64)
Docker for mac 4.10.1 (not running)

### Issue
Crash when launching a job.

### Reproduction steps
run `$ nomad agent -dev > nomad.out 2>&1 &`
run `nomad job status` to verify nomad is running.
run `$ nomad job run https.nomad`

#### Expected Result
without docker running I would expect the job to fail to schedule the job. 
with docker running I expect the job to run.

#### Actual Result
`$ nomad job run https.nomad`
`Error submitting job: Put "http://127.0.0.1:4646/v1/jobs": EOF`

### Job file (if appropriate)
my sample job file....
```hcl
job "https" {
  datacenters = ["dc1"]
  type        = "system"

  group "echo" {
    network {
      mode = "host"
      port "http" {
        static = 80
      }
    }

    task "redirect" {
      driver = "docker"

      config {
        image        = "vibrato/https-echo:v0.0.6"
        args = [
          "-listen", ":${NOMAD_PORT_http}",
        ]
        ports = ["http"]
      }
    }
  }
}
```

### Nomad Server logs (if appropriate)
[nomad output](https://gist.github.com/tristanmorgan/eb24dbf1ffb60121497a848afacd6ec8)

### Nomad Client logs (if appropriate)
[the same nomad output](https://gist.github.com/tristanmorgan/eb24dbf1ffb60121497a848afacd6ec8)

**Comment by tristanmorgan** [comment]
It seems the Enterprise binary is not affected by this. **just** the OSS Version.

- `Nomad v1.3.2+ent (c0cc8626b2284ff7cb8e2aaa2d7d6943bb376574)` is OK.
- `Nomad v1.3.1 (2b054e38e91af964d1235faa98c286ca3f527e56)` is also OK.

**Comment by tgross** [comment]
Hi @tristanmorgan! One of my colleagues started looking into this and it looks like it's a problem specific to the build environment for M1 on GitHub Actions, as it'll reproduce every time we try to build that specific version of Nomad there, but not on the M1 developer laptops a few of our team have. We're still investigating but in the meantime as a workaround you should be able to build from source without trouble as far as we know. And if you don't find that works, please let us know because that might give us more clues as to the problem!

I've tagged @lgfa29 as the investigator on this but he may need to hand off to someone else. We'll update here once we know more.

**Comment by lgfa29** [comment]
After some investigation I still don't have a clear view of that could be the root cause. Changing some things in the environment (like updating the [macOS](https://github.com/hashicorp/nomad/actions/runs/2719303997) or [Go](https://github.com/hashicorp/nomad/commit/a0ea6957c34a449f8d10c95e6ec3a29123e1d9fa) version used to build) generates a binary that doesn't trigger this panic.

I also haven't been able to generate a binary that panics locally, but it's my current line of investigation.

**Comment by tristanmorgan** [comment]
Looks like it is fixed in the 1.3.3 build.
closing.

**Comment by lgfa29** [comment]
Thank you for confirming 1.3.3 is working. We haven't been able to determined the root cause of the issue in 1.3.2 but we noticed that upgrading Go fixed the problem, so we did that for 1.3.3 🙂
